Source/WTF/wtf/StdLibExtras.h

```diff
@@ -671,6 +671,51 @@ template<typename OptionalType> auto valueOrDefault(OptionalType&& optionalValue
     return optionalValue ? *std::forward<OptionalType>(optionalValue) : std::remove_reference_t<decltype(*optionalValue)> { };
 }
 
+template<typename T, typename U, std::size_t Extent>
+std::span<T, Extent> spanReinterpretCast(std::span<U, Extent> span)
+{
+    RELEASE_ASSERT(!(span.size_bytes() % sizeof(T)));
+    static_assert(std::is_const_v<T> || (!std::is_const_v<T> && !std::is_const_v<U>), "spanCast will not remove constness from source");
+    return std::span<T, Extent> { reinterpret_cast<T*>(const_cast<std::remove_const_t<U>*>(span.data())), span.size_bytes() / sizeof(T) };
+}
+
+template<typename T, std::size_t Extent>
+std::span<const uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> asBytes(std::span<T, Extent> span)
+{
+    return std::span<const uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> { reinterpret_cast<const uint8_t*>(span.data()), span.size_bytes() };
+}
+
+template<typename T, std::size_t Extent>
+std::span<uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> asWritableBytes(std::span<T, Extent> span)
+{
+    return std::span<uint8_t, Extent == std::dynamic_extent ? std::dynamic_extent: Extent * sizeof(T)> { reinterpret_cast<uint8_t*>(span.data()), span.size_bytes() };
+}
+
+template<typename T, std::size_t Extent1, std::size_t Extent2>
+bool equalSpans(std::span<T, Extent1> a, std::span<T, Extent2> b)
+{
+    if (a.size() != b.size())
+        return false;
+    return !memcmp(a.data(), b.data(), a.size() * sizeof(T));
+}
+
+template<typename T, std::size_t TExtent, typename U, std::size_t UExtent>
+void memcpySpan(std::span<T, TExtent> destination, std::span<U, UExtent> source)
+{
+    RELEASE_ASSERT(destination.size() == source.size());
+    static_assert(sizeof(T) == sizeof(U));
+    static_assert(std::is_trivially_copyable_v<T>);
+    static_assert(std::is_trivially_copyable_v<U>);
+    memcpy(destination.data(), source.data(), destination.size_bytes());
+}
+
+template<typename T, std::size_t Extent>
+void memsetSpan(std::span<T, Extent> destination, uint8_t byte)
+{
+    static_assert(std::is_trivially_copyable_v<T>);
```

Comment:
Is `std::is_trivially_copyable_v` enough to cover a types with non-trivial destructors or constructors?

Reply:
Per https://en.cppreference.com/w/cpp/language/classes#Trivially_copyable_class:
- each eligible copy constructor is trivial
- has a non-deleted trivial destructor

https://en.cppreference.com/w/cpp/string/byte/memset also mentions that behavior for non-trivially copyable types is undefined.